activesupport/lib/active_support/deprecation/instance_delegator.rb

```diff
@@ -18,7 +17,29 @@ def include(included_module)
         end
 
         def method_added(method_name)
-          singleton_class.delegate(method_name, to: :instance)
+          use_instead =
+            case method_name
+            when :silence, :behavior=, :disallowed_behavior=, :disallowed_warnings=, :silenced=, :debug=
+              target = "(defined?(Rails.application.deprecators) ? Rails.application.deprecators : ActiveSupport::Deprecation._instance)"
+              "Rails.application.deprecators.#{method_name}"
+            when :warn, :deprecate_methods, :gem_name, :gem_name=, :deprecation_horizon, :deprecation_horizon=
+              "your own Deprecation object"
+            else
+              "Rails.application.deprecators[framework].#{method_name} where framework is for example :active_record"
+            end
+          args = /[^\]]=\z/.match?(method_name) ? "arg" : "..."
+          target ||= "ActiveSupport::Deprecation._instance"
+          singleton_class.module_eval <<~RUBY, __FILE__, __LINE__ + 1
+            def #{method_name}(#{args})
+              ActiveSupport.deprecator.warn("Calling #{method_name} on ActiveSupport::Deprecation is deprecated and will be removed from Rails (use #{use_instead} instead)")
+              #{target}.#{method_name}(#{args})
+            end
+          RUBY
+        end
+
+        def instance
+          ActiveSupport.deprecator.warn("ActiveSupport::Deprecation.instance is deprecated (use your own Deprecation object)")
+          super
```

Comment:
Similarly, to cover e.g. `ActiveSupport::Deprecation.instance.behavior = ...`:

```suggestion
          super
          ensure
            ActiveSupport::Deprecation._instance.warn("ActiveSupport::Deprecation.instance is deprecated (use your own Deprecation object)")
```

Or, if we make `ActiveSupport.deprecator` == `ActiveSupport::Deprecation._instance`:

```suggestion
          super
          ensure
            ActiveSupport.deprecator.warn("ActiveSupport::Deprecation.instance is deprecated (use your own Deprecation object)")
```

Reply:
You're right — swapping the order here is nonsensical. :man_facepalming: :man_facepalming:

Using `ActiveSupport::Deprecation._instance` as the underlying deprecator will benefit subsequent calls though.  e.g.:

```ruby
ActiveSupport::Deprecation.instance.behavior = ...
ActiveSupport::Deprecation.instance.disallowed_behavior = ...
```

(Which is now covered since `ActiveSupport.deprecator` is `ActiveSupport::Deprecation._instance`.)